api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java

```diff
@@ -503,17 +468,20 @@ public Long findSurveyCount(String name, String term) {
   }
 
   private String modifyTermMatch(String term) {
-    if (!MYSQL_FULL_TEXT_CHARS.stream()
-        .filter(term::contains)
-        .collect(Collectors.toList())
-        .isEmpty()) {
-      return term;
+    if (MYSQL_FULL_TEXT_CHARS.stream().anyMatch(term::contains)) {
+      return Arrays.stream(term.split("\\s+"))
+          .map(
+              s -> {
+                if (s.startsWith("(")
+                    || (!s.startsWith("+") && !s.startsWith("-") && !s.endsWith(")"))) {
+                  return "+" + s;
+                }
+                return s;
+              })
+          .collect(Collectors.joining(" "));
     }
 
     String[] keywords = term.split("\\W+");
-    if (keywords.length == 1 && keywords[0].length() <= 3) {
```

Comment:
Why was this removed?

Reply:
This was mainly to handle exact matches on code, but the UI no longer allows words under 3 characters to be submitted and the dao checks codes with a like in the dao: https://github.com/all-of-us/workbench/blob/3f6a918b1ddc56fa32f0748d05159ad3e2858030/api/src/main/java/org/pmiops/workbench/cdr/dao/CBCriteriaDao.java#L159-L159